src/components/DefaultVideo/DefaultVideo.tsx

```diff
@@ -0,0 +1,106 @@
+import React, {Fragment, useCallback, useImperativeHandle, useRef, useState} from 'react';
+
+import {CustomControlsType, MediaVideoControlsType, MediaVideoProps} from '../../models';
+import {block} from '../../utils';
+import {getVideoTypesWithPriority} from '../Media/Video/utils';
+import CustomBarControls from '../ReactPlayer/CustomBarControls';
+
+import './DefaultVideo.scss';
+
+const b = block('default-video');
+
+type DefaultVideoRefType = HTMLVideoElement | undefined;
+
+interface DefaultVideoProps {
+    video: MediaVideoProps;
+    qa?: string;
+    customBarControlsClassName?: string;
+    className?: string;
+}
+
+export const DefaultVideo = React.forwardRef<DefaultVideoRefType, DefaultVideoProps>(
+    (props, ref) => {
+        const {video, qa, customBarControlsClassName} = props;
+        const {controls, customControlsOptions, muted: initiallyMuted} = video;
+        const {
+            muteButtonShown,
+            positioning,
+            type: customControlsType,
+        } = customControlsOptions || {};
+        const [isPaused, setIsPaused] = useState(false);
+        const [isMuted, setIsMuted] = useState(initiallyMuted);
+        const videoRef = useRef<HTMLVideoElement>(null);
+
+        useImperativeHandle(
```

Comment:
what is the point of `useImperativeHandle` usage here?

Reply:
added comment